newrelic/node-newrelic
# Added ability to register custom ESM instrumentation

## Proposed Release Notes
 * Added a configuration option(`api.esm.custom_instrumentation_entrypoint`) to set an entrypoint for custom ESM instrumentation registration.

## Links
Closes NEWRELIC-3778

## Details
You will have to test this locally by running `npm ci --force` to install and force eslint peer dep resolution. Tracking the fix [here](https://issues.newrelic.com/browse/NEWRELIC-4168)


You can see a working example [here](https://github.com/newrelic/newrelic-node-examples/tree/esm-app).  

To test:
 1. Clone this branch and run `npm link`
 2. Clone the branch of the example app and run `npm link newrelic`
 3. Follow README and you should see custom instrumentation loading and working 🎉 

I did not update the versioned tests and it's a little weird how we have a test loader that injects a mock agent and i couldn't control the order to properly set a config to register the custom entrypoint but I did update unit tests that were really integration to prove this method will work
